packages/react/src/template.tsx

```diff
@@ -18,13 +18,14 @@ export type DataState<T = EventDataState> =
   | ReferencedComponentDataState<T>
   | ComponentDataState<T>
   | ReferencedDataState
-  | ActionDataState
+  | HandlerDataState<T>
   | StateDataState
   | { [key: string]: DataState<T>; $?: never }
-  | DataState[]
+  | DataState<T>[]
   | T
 /** Server data state */
 export type ServerDataState = DataState<ServerEventDataState>
+export type ServerHandlerDataState = HandlerDataState<ServerEventDataState>
```

Comment:
This is just all types that are valid to be passed where component expects a function.